health/handler.go

```diff
@@ -0,0 +1,107 @@
+package health
+
+import (
+	"encoding/json"
+	"fmt"
+	gosundheit "github.com/AppsFlyer/go-sundheit"
+	"github.com/openziti/fabric/xweb"
+	"github.com/sirupsen/logrus"
+	"net/http"
+	"strings"
+	"time"
+)
+
+const (
+	Binding = "health-checks"
+)
+
+var _ xweb.WebHandlerFactory = &HealthCheckApiFactory{}
+
+func NewHealthCheckApiFactory(healthChecker gosundheit.Health) *HealthCheckApiFactory {
+	return &HealthCheckApiFactory{
+		healthChecker: healthChecker,
+	}
+}
+
+type HealthCheckApiFactory struct {
+	healthChecker gosundheit.Health
+}
+
+func (factory HealthCheckApiFactory) Validate(*xweb.Config) error {
+	return nil
+}
+
+func (factory HealthCheckApiFactory) Binding() string {
+	return Binding
+}
+
+func (factory HealthCheckApiFactory) New(_ *xweb.WebListener, options map[interface{}]interface{}) (xweb.WebHandler, error) {
+	return &HealthCheckApiHandler{
+		healthChecker: factory.healthChecker,
+		options:       options,
+	}, nil
+}
+
+type HealthCheckApiHandler struct {
+	options       map[interface{}]interface{}
+	healthChecker gosundheit.Health
+}
+
+func (self *HealthCheckApiHandler) Binding() string {
+	return Binding
+}
+
+func (self *HealthCheckApiHandler) Options() map[interface{}]interface{} {
+	return self.options
+}
+
+func (self *HealthCheckApiHandler) RootPath() string {
+	return "/health-checks"
+}
+
+func (self *HealthCheckApiHandler) IsHandler(r *http.Request) bool {
+	return strings.HasPrefix(r.URL.Path, self.RootPath())
+}
+
```

Comment:
@andrewpmartinez tried to make this match the edge API, so output looks like:

```
{
    "data": {
        "checks": [
            {
                "healthy": true,
                "id": "bolt.read",
                "lastCheckDuration": "19.699µs",
                "lastCheckTime": "2021-06-18T21:48:37-04:00"
            }
        ],
        "healthy": true
    },
    "meta": {}
}
```
Let me know that works.

Reply:
If this is a list, `data` should be the array. If this is a single object that is returned and there are other properties other than `checks` it is fine.

Reply:
There's a `healthy` which provides a quick path to the overall health at the same level as `checks`

Reply:
Just noticed your `lastCheckedTime` has a TZ. I think we try to output in UTC. Not a huge deal though.